config/config/src/index.ts

```diff
@@ -291,7 +291,7 @@ export async function getConfig (
     ...rcOptions.map((configKey) => [camelcase(configKey), npmConfig.get(configKey)]) as any, // eslint-disable-line
     ...Object.entries(cliOptions).filter(([name, value]) => typeof value !== 'undefined').map(([name, value]) => [camelcase(name), value]),
   ]) as unknown as ConfigWithDeprecatedSettings
-  const cwd = betterPathResolve(cliOptions.dir ?? npmConfig.localPrefix)
+  const cwd = await fs.promises.realpath(betterPathResolve(cliOptions.dir ?? npmConfig.localPrefix))
```

Comment:
You can make this operation synchronous as we don't run anything concurrently at this point.

Reply:
I know for a fact you are already asynchronous [really early](https://github.com/pnpm/pnpm/blob/5a5e42551e81f7ebca04b9eaabba854ae5fd75ed/config/config/src/index.ts#L192) in this function.
Also you do not start additional tasks so it should not be a problem.

Reply:
That can be changed too. But not in this PR. Sync fs operations in Node.js are faster unless you can run them in parallel to something else.